<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1349</width>
    <height>830</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="Robot_Control">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>290</y>
      <width>521</width>
      <height>201</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Velocity</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_7">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>496</width>
        <height>155</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_59">
           <property name="text">
            <string>LinearStep</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="linear_velocity_step"/>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_60">
           <property name="text">
            <string>AngularStep</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="angular_velocity_step"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QPushButton" name="Vx_p">
             <property name="text">
              <string>Vx+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="current_vec_x"/>
           </item>
           <item>
            <widget class="QPushButton" name="Vx_n">
             <property name="text">
              <string>Vx-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QPushButton" name="Vy_p">
             <property name="text">
              <string>Vy+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="current_vec_y"/>
           </item>
           <item>
            <widget class="QPushButton" name="Vy_n">
             <property name="text">
              <string>Vy-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QPushButton" name="Vz_p">
             <property name="text">
              <string>Vz+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="current_vec_z"/>
           </item>
           <item>
            <widget class="QPushButton" name="Vz_n">
             <property name="text">
              <string>Vz-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QPushButton" name="Wx_p">
             <property name="text">
              <string>Wx+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="current_vec_rx"/>
           </item>
           <item>
            <widget class="QPushButton" name="Wx_n">
             <property name="text">
              <string>Wx-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QPushButton" name="Wy_p">
             <property name="text">
              <string>Wy+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="current_vec_ry"/>
           </item>
           <item>
            <widget class="QPushButton" name="Wy_n">
             <property name="text">
              <string>Wy-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QPushButton" name="Wz_p">
             <property name="text">
              <string>Wz+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="current_vec_rz"/>
           </item>
           <item>
            <widget class="QPushButton" name="Wz_n">
             <property name="text">
              <string>Wz-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_14">
         <item>
          <widget class="QPushButton" name="MoveToZero">
           <property name="text">
            <string>MoveToZero</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="Stop">
           <property name="text">
            <string>Stop</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Position</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_11">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>20</y>
        <width>496</width>
        <height>131</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_17">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QLabel" name="label_61">
             <property name="text">
              <string>LinearStep</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="linear_position_step"/>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <item>
            <widget class="QLabel" name="label_62">
             <property name="text">
              <string>AngularStep</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="angular_position_step"/>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_11">
           <item>
            <widget class="QPushButton" name="Px_p">
             <property name="text">
              <string>X+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="Px_n">
             <property name="text">
              <string>X-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_13">
           <item>
            <widget class="QPushButton" name="Py_p">
             <property name="text">
              <string>Y+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="Py_n">
             <property name="text">
              <string>Y-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_12">
           <item>
            <widget class="QPushButton" name="Pz_p">
             <property name="text">
              <string>Z+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="Pz_n">
             <property name="text">
              <string>Z-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_14">
           <item>
            <widget class="QPushButton" name="Rx_p">
             <property name="text">
              <string>Rx+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="Rx_n">
             <property name="text">
              <string>RX-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_18">
           <item>
            <widget class="QPushButton" name="Ry_p">
             <property name="text">
              <string>Ry+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="Ry_n">
             <property name="text">
              <string>Ry-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_15">
           <item>
            <widget class="QPushButton" name="Rz_p">
             <property name="text">
              <string>Rz+</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="Rz_n">
             <property name="text">
              <string>Rz-</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_18">
         <item>
          <widget class="QPushButton" name="MoveToZero_2">
           <property name="text">
            <string>MoveToZero</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="Stop_2">
           <property name="text">
            <string>Stop</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_8">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>160</y>
      <width>451</width>
      <height>118</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_8">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QPushButton" name="connect">
         <property name="text">
          <string>ConnectRobot</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="disconnect">
         <property name="text">
          <string>DisconnectRobot</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QPushButton" name="StartUpdate">
         <property name="text">
          <string>StartUpdate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="StopUpdate">
         <property name="text">
          <string>StopUpdate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="getState">
         <property name="text">
          <string>GetState</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="FindReferance">
         <property name="text">
          <string>FindReferance</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_15">
       <item>
        <widget class="QLineEdit" name="current_pos_x"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_pos_y"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_pos_z"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_pos_rx"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_pos_ry"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_pos_rz"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_17">
       <item>
        <widget class="QLineEdit" name="current_joint_pos_1"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_joint_pos_2"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_joint_pos_3"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_joint_pos_4"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_joint_pos_5"/>
       </item>
       <item>
        <widget class="QLineEdit" name="current_joint_pos_6"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_9">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>60</y>
      <width>301</width>
      <height>92</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_9">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QPushButton" name="pushButton_Connect">
         <property name="text">
          <string>ConnectCamera</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_StartCamera">
         <property name="text">
          <string>StartCamera</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Exposure</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_expose"/>
       </item>
       <item>
        <widget class="QPushButton" name="expose">
         <property name="font">
          <font>
           <pointsize>12</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Set</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QSlider" name="horizontalSliderExpose">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_SystemStatus">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>281</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>SystemStatus</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_RobotStatus">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>110</y>
      <width>81</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Robot Status</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_CameraStatus">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>60</y>
      <width>111</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Camera Status</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>30</y>
      <width>491</width>
      <height>26</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_11">
     <item>
      <widget class="QPushButton" name="pushButton_Start">
       <property name="text">
        <string>Start</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_Stop">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Stop</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radioButton_VisualServoing">
       <property name="text">
        <string>VisualServoing</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radioButton_Sharpness">
       <property name="text">
        <string>Sharpness</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radioButton_Calibration">
       <property name="text">
        <string>Calibration</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="SaveDesiredImage">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>500</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>SaveDesiredImage </string>
    </property>
   </widget>
   <widget class="QLabel" name="label_pic">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>319</width>
      <height>237</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">
border-top-color: rgb(85, 255, 0);</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_pic_desired">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>300</y>
      <width>319</width>
      <height>237</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border-color: rgb(170, 85, 0);
border-width: 2px</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>540</y>
      <width>851</width>
      <height>241</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_12">
     <item>
      <widget class="QChartView" name="charts_feature_error" native="true">
       <property name="font">
        <font>
         <family>Arial</family>
        </font>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QChartView" name="charts_linear_velocity" native="true">
       <property name="font">
        <font>
         <family>Arial</family>
        </font>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QChartView" name="charts_angular_velocity" native="true">
       <property name="font">
        <font>
         <family>Arial</family>
        </font>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>70</y>
      <width>181</width>
      <height>24</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_13">
     <item>
      <widget class="QLabel" name="label_CameraStatus_2">
       <property name="text">
        <string>Sharpness</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="SharpnessValue">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="margin">
        <number>2</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QChartView" name="charts_sharpness" native="true">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>120</y>
      <width>280</width>
      <height>239</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
     </font>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1349</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>显微视觉伺服</string>
    </property>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QChartView</class>
   <extends>QWidget</extends>
   <header>qchartview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
